analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicRuleCheck.cs

```diff
@@ -46,7 +49,17 @@ public void Init(SonarAnalysisContext sonarContext, SyntaxNodeAnalysisContext no
             NodeContext = nodeContext;
         }
 
-        protected void ReportIssue(IOperationWrapperSonar operation) =>
-            NodeContext.ReportIssue(Diagnostic.Create(Rule, operation.Instance.Syntax.GetLocation()));
+        protected void ReportIssue(IOperationWrapperSonar operation, params object[] messageArgs) =>
+            ReportIssue(operation.Instance, messageArgs);
+
+        protected void ReportIssue(IOperation operation, params object[] messageArgs)
+        {
+            var location = operation.Syntax.GetLocation();
+            if (!reportedDiagnostics.Contains(location))
```

Comment:
`Add` returns `bool` so we can simplify this and remove the other `Add`
```suggestion
            if (reportedDiagnostics.Add(location))
```